scripts/build-ancillary-repository.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+#
+# Copyright 2018 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+TOP=$(git rev-parse --show-toplevel 2>/dev/null)
+
+if [[ -z "$TOP" ]]; then
+	echo "Must be run inside the git repsitory." 2>&1
+	exit 1
+fi
+
+. "$TOP/scripts/functions.sh"
+
+set -o xtrace
+set -o errexit
+set -o pipefail
+
+function resolve_s3_uri()
```

Comment:
I feel like the logic where it would determine automatically whether a URI or PREFIX was passed was better. Otherwise we can't currently pass a URI from the jenkins job.
We can just check whichever of URI/PREFIX is non empty and pass that one to `resolve_s3_uri`, and revert to the old logic.

Reply:
well, we need to update Jenkins to use URI.. we should have a path forward to make that change, without requiring a flag day.

so if you don't want to make this change right now, I think it'll eventually be needed.

Reply:
```
--- a/scripts/build-ancillary-repository.sh
+++ b/scripts/build-ancillary-repository.sh
@@ -39,6 +39,8 @@ function resolve_s3_uri()

    if [[ -n "$pkg_uri" ]]; then
        local resolved_uri="$pkg_uri"
+   elif [[ "$pkg_prefix" == s3* ]]; then
+       local resolved_uri="$pkg_prefix"
    elif [[ -n "$pkg_prefix" ]]; then
        local resolved_uri="s3://$bucket/$pkg_prefix"
```
I added the above to make it easy to use this with the current jenkins parameters.. that addresses your concern right?